src/d3a/models/market/one_sided.py

```diff
@@ -117,14 +123,22 @@ def accept_offer(self, offer_or_id: Union[str, Offer], buyer: str, *, energy: in
 
                     if trade_rate is None:
                         trade_rate = offer.price / offer.energy
+
                     # TODO: Remove the math.floor from the lower threshold once the pay-as-clear
                     # market is refactored (D3ASIM-907)
                     assert trade_rate + FLOATING_POINT_TOLERANCE >= \
                         math.floor(offer.price / offer.energy)
 
+                    # reducing trade_rate to be charged in terms of grid_fee
+                    if iaa_fee:
+                        source_rate = trade_rate / (1 + self.transfer_fee_ratio)
```

Comment:
Attention!   a / (1+b) != a - a*b 
I guess you were looking for `source_rate = trade_rate * (1 - self.transfer_fee_ratio)`, right?

Reply:
https://github.com/gridsingularity/d3a/pull/400/files#r267679532